### PR TITLE
Fix yarn id after yarn restart

### PR DESCRIPTION
**Pull request by Minutis**

### CODE REVIEW

File: server/src/main/java/com/exacaster/lighter/backend/yarn/YarnBackend.java [hunk 1]
```diff
@@ -121,19 +122,16 @@ public Map<String, String> getSubmitConfiguration(Application application,
     }
 
     private Optional<String> getYarnApplicationId(Application application) {
-        return Optional.ofNullable(application.getAppId())
```
Comment (pdambrauskas) [inline server/src/main/java/com/exacaster/lighter/backend/yarn/YarnBackend.java:124]: Why this change was needed? Is AppId always present?

Comment (pdambrauskas) [reply server/src/main/java/com/exacaster/lighter/backend/yarn/YarnBackend.java:124]: Oh, I got it.

**Comment by pdambrauskas** [comment]
Are Application ids changing when yarn restart, can you elaborate more on your fix?

**Comment by Minutis** [comment]
> Are Application ids changing when yarn restart, can you elaborate more on your fix?
After YARN restart all IDs are restarted. If you had `application_xxxxxxxxx_51` and YARN got restarted you immediately get errors. Because we are using the ID that we've stored in Lighter's DB to fetch states and etc.

**Comment by pdambrauskas** [comment]
> After YARN restart all IDs are restarted.

Does not sound right, are you saying that jobs are not interrupted when YARN server is restarted and their IDs are somehow updated? What about logs of these jobs do we lose them?

**Comment by Minutis** [comment]
> > After YARN restart all IDs are restarted.
> 
> Does not sound right, are you saying that jobs are not interrupted when YARN server is restarted and their IDs are somehow updated? What about logs of these jobs do we lose them?

As far as I know, we lose logs after the YARN restart. At least from my practice after restarting the YARN resource manager and opening the UI you see an empty YARN as if not a single application was ever run there.

**Comment by pdambrauskas** [comment]
> Running jobs fails and "are gone" if YARN is restarted.

So, then why this change is needed? In what situation we have non-blank appId, which does not match with real appId?

**Comment by Minutis** [comment]
> > Running jobs fails and "are gone" if YARN is restarted.
> 
> So, then why this change is needed? In what situation we have non-blank appId, which does not match with real appId?

We save appId in our DB which if YARN was restarted becomes invalid. We have a record in DB that points to let's say to `application_xxxxxxxxx_51` appId, but in YARN this ID no longer exists.

My solution to this is to always fetch the ID from YARN, so we have latest data and do not rely in appId savid in our DB.

**Comment by Minutis** [comment]
We had a small live discussion regarding this. I will close this MR and make a new one.

The idea behind closing this MR is that appId that's in the database is correct just not present in YARN. So failure should happen not when fetching appId from the YARN but whenever some data is fetched from YARN using the persisted appId.
